client/src/main/java/sigblockchain/projectred/client/Block.java

```diff
@@ -0,0 +1,82 @@
+package sigblockchain.projectred.client;
+
+public class Block {
+    private final int Version;          // uint16
+    private final long Height;          // uint64
+    private final long Timestamp;       // int64
+    private final byte[] PreviousHash;  // []byte
+    private final byte[] MerkleRootHash;// []byte
+    private final int DataLen;          // uint16
+    private final byte[][] Data;        // [][]byte
+
+    public Block(int version, long height, long timestamp, byte[] previousHash, byte[] merkleRootHash, int dataLen, byte[][] data) {
+        if (!checkCompatibility(version, height, timestamp, dataLen))
+            throw new IllegalArgumentException("Invalid Input Values");
+
+//        //Individual values Checking
+//        if (!checkUnsigned16(version))
+//            throw new IllegalArgumentException("Invalid Input version");
+//        if (!checkUnsigned64(height))
+//            throw new IllegalArgumentException("Invalid Input height");
+//        if (!checkSigned64(timestamp))
+//            throw new IllegalArgumentException("Invalid Input timeStamp");
```

Comment:
1. Don't commit large blocks of commented code. 
2. I think it would be nice to display which variable gave the exception.

Reply:
Fixed!